tasks/main.yml

```
@@ -106,6 +107,38 @@
   tags:
     - wg-config
 
+- name: Create private key for unmanaged hosts
+  shell: "wg genkey | tee {{ wireguard_remote_directory }}/{{ item.host }}-privatekey"
+  args:
+    creates: "{{ wireguard_remote_directory }}/{{ item.host }}-privatekey"
+  register: uh_privkey
+  with_items: "{{ wireguard_unmanaged_hosts | default([]) }}"
+
+- name: Validate permissions of unmanaged hosts' private keys
+  file:
+    path: "{{ wireguard_remote_directory }}/{{ item.host }}-privatekey"
+    mode: '0400'
+  with_items: "{{ wireguard_unmanaged_hosts | default([]) }}"
+
+- name: Recover existing private key for unmanaged hosts
+  shell: "cat {{ wireguard_remote_directory }}/{{ item.host }}-privatekey"
+  register: uh_privkey
+  changed_when: false
+  with_items: "{{ wireguard_unmanaged_hosts | default([]) }}"
+
+- name: Derive WireGuard public key for unmanaged hosts
+  shell: "cat {{ wireguard_remote_directory }}/{{ item.host }}-privatekey | wg pubkey | tee {{ wireguard_remote_directory }}/{{ item.host }}-pubkey"
+  args:
+    creates: "{{ wireguard_remote_directory }}/{{ item.host }}-pubkey"
+  register: uh_pubkey
```

```suggestion
  register: uh_public_key
```
Not need to shorten variable name.